sdk/core/src/retry_policy.rs

```diff
@@ -0,0 +1,123 @@
+use tokio::time::{sleep, Duration};
+
+#[async_trait::async_trait]
+pub trait RetryPolicy {
+    async fn create_instance(&self) -> Box<dyn RetryPolicyInstance>;
+}
+
+#[async_trait::async_trait]
+pub trait RetryPolicyInstance {
+    async fn retry(&mut self) -> bool;
```

Comment:
nit: the name `retry` makes it sound like this is in charge of retrying. Perhaps `should_retry` would be clearer?